app/src/main/java/bankingapp/App.java

```diff
@@ -39,18 +40,65 @@ public static void main(String[] args) {
                             "0. Exit");
 
                     try {
-                        response = input.nextInt();
+                        response = Integer.parseInt(input.next());
                         switch (response) {
+
                             case 1:
-                                System.out.println(new CreateAccount(connection));
+                                //Create a new user account
+                                String firstName;
+                                String lastName;
+
+                                 //Prompt user input for first and last names
+                                 System.out.println("Enter First Name: ");
+                                 do {
+                                    firstName = input.nextLine();
+                                 } while (firstName.equals(null) || firstName.trim().isEmpty());
+
+                                 System.out.println("Enter Last Name: ");
+                                 do {
+                                    lastName = input.nextLine();
+                                 }while (lastName.equals(null) || lastName.trim().isEmpty());
```

Comment:
This is much better! It helps isolate the CreateAccount class from the UI, as we discussed.

As you're improving things, keep an eye out for lines where you find yourself repeating the same code multiple times, as here, where you're repeating largely identical code for both firstName and lastName. This code is ripe for refactoring into a helper method! The parts which are different can become parameters to the new function.

Can you write the `askUserForString` method so that the following code works?

```
case 1:
    String firstName = askUserForString("First Name");
    String lastName = askUserForString("Last Name");

   CreateAccount account = new CreateAccount();
   // etc
```

Reply:
This is an application of the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle, which gives a lot of bang for the buck, even in small cases like this:

* By repeating code, you double the chance of bugs in your code.
* A future reader of the existing code will have to re-read the same code for last name as for first name; whereas a reader of my suggested code would only need to read `askUserForString` once.
* A future modifier of the existing code may forget to modify the logic for both first name and last name; or they may make different changes to each, and forget which one is correct.